pkg/kds/server/server_test.go

```diff
@@ -0,0 +1,628 @@
+package server_test
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	v2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	envoy_core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/duration"
+	"github.com/golang/protobuf/ptypes/wrappers"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+
+	mesh_proto "github.com/Kong/kuma/api/mesh/v1alpha1"
+	kuma_cp "github.com/Kong/kuma/pkg/config/app/kuma-cp"
+	kds_config "github.com/Kong/kuma/pkg/config/kds"
+	"github.com/Kong/kuma/pkg/core"
+	"github.com/Kong/kuma/pkg/core/resources/apis/mesh"
+	"github.com/Kong/kuma/pkg/core/resources/manager"
+	"github.com/Kong/kuma/pkg/core/resources/model"
+	"github.com/Kong/kuma/pkg/core/resources/store"
+	"github.com/Kong/kuma/pkg/core/runtime"
+	"github.com/Kong/kuma/pkg/core/runtime/component"
+	"github.com/Kong/kuma/pkg/kds"
+	kds_server "github.com/Kong/kuma/pkg/kds/server"
+	mads_server "github.com/Kong/kuma/pkg/mads/server"
+	"github.com/Kong/kuma/pkg/plugins/resources/memory"
+	util_xds "github.com/Kong/kuma/pkg/util/xds"
+)
+
+var (
+	node = &envoy_core.Node{
+		Id:      "test-id",
+		Cluster: "test-cluster",
+	}
+)
+
+const (
+	defaultTimeout = 3 * time.Second
+)
+
+type mockStream struct {
```

Comment:
I'd be surprised if there is not already a test library to mock streams. In any case, I think this code is worth separating in its own file. It can be in this folder for now, but let's just keep it separate.